doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -777,42 +740,46 @@ type which is the type of its body.
 A formal presentation of constants and environments is given in
 Section :ref:`typing-rules`.
 
-.. cmd:: Definition @ident := @term
+.. cmd:: {| Definition | Example } @ident_decl @def_body
+   :name: Definition; Example
+
+   .. insertprodn def_body def_body
+
+   .. prodn::
+      def_body ::= {* @binder } := {? @reduce } @term
+      | {* @binder } : @type := {? @reduce } @term
+      | {* @binder } : @type
+
+   These commands bind :n:`@term` to the name :n:`@ident` in the environment,
+   provided that :n:`@term` is well-typed.  They can take the :attr:`local` attribute (see :ref:`gallina-attributes`),
+   which makes the defined :n:`@ident` accessible by :cmd:`Import` and its variants
+   only through their fully qualified names.
+   If :n:`@reduce` is present then :n:`@ident` is bound to the result of the specified
+   computation on :n:`@term`.
+
+   If :n:`@term` is omitted, Coq enters the proof editing mode.  This is useful WHY EXACTLY?
```

Comment:
This can be used to define a term incrementally, in particular by relying on the `refine` tactic.